### PR TITLE
Add documentation for opam-monorepo

### PR DESCRIPTION
**Pull request by emillon**
This contains a FAQ, a description of the concepts and files, and some workflows.

It is integrated with `dune build @doc` but it is probably up to us to build and host the generated HTML.

### CODE REVIEW

**Review by NathanReb — COMMENTED** [review]
Looks great!

I suspect we'll be more inclined to write documentation now that we have some!

File: doc/faq.mld [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{1 FAQ}
+
+{2 Is This For Me?}
+
+This might be the right thing for you if all your dependencies use Dune and you are under full control of the build and deployment pipeline for your piece of software (e.g., if you are in a position to write a [dune-workspace] file).
```
[the remark this answers is not in the document]
Comment (emillon) [reply doc/faq.mld:5]: These days I think that hard wrapping markdown is a distraction (it makes suggestions difficult for example) but happy to rewrap.

[the remark this answers is not in the document]
Comment (NathanReb) [reply doc/faq.mld:5]: The main problem I have with this is it makes review a bit harder as you have to scroll horizontally.

For suggestions I think multiline suggestions can be used to work around wrapping problems. It's not ideal but it's decent!

[the remark this answers is not in the document]
Comment (emillon) [reply doc/faq.mld]: OK, I wrapped everything.

**Pull request merged**